src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/SetActivationDeploymentsPartialValidationRule.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Threading.Tasks;
+using Stratis.Bitcoin.Consensus.Rules;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules.CommonRules
+{
+    /// <summary>Set the <see cref="RuleContext.Flags"/> property that defines what deployments have been activated.</summary>
+    public class SetActivationDeploymentsPartialValidationRule : PartialValidationConsensusRule
+    {
+        /// <inheritdoc />
+        /// <exception cref="ConsensusErrors.InvalidPrevTip">The tip is invalid because a reorg has been detected.</exception>
+        public override Task RunAsync(RuleContext context)
+        {
+            // Calculate the consensus flags and check they are valid.
+            context.Flags = this.Parent.NodeDeployments.GetFlags(context.ValidationContext.ChainTipToExtend);
+
+            return Task.CompletedTask;
+        }
+    }
+
+    /// <summary>Set the <see cref="RuleContext.Flags"/> property that defines what deployments have been activated.</summary>
+    public class SetActivationDeploymentsFullValidationRule : FullValidationConsensusRule
```

Comment:
I don't think you need this duplication of the code

Reply:
We kind of do because this rule is both for PV and FV. )